erpnext/patches/v12_0/unset_customer_supplier_based_on_type_of_item_price.py

```diff
@@ -1,15 +1,29 @@
 from __future__ import unicode_literals
 import frappe
 
+
 def execute():
-    invalid_selling_item_price = frappe.db.sql(
-        """SELECT name FROM `tabItem Price` WHERE selling = 1 and buying = 0 and (supplier IS NOT NULL or supplier = '')"""
-    )
-    invalid_buying_item_price = frappe.db.sql(
-        """SELECT name FROM `tabItem Price` WHERE selling = 0 and buying = 1 and (customer IS NOT NULL or customer = '')"""
-    )
-    docs_to_modify = invalid_buying_item_price + invalid_selling_item_price
-    for d in docs_to_modify:
-        # saving the doc will auto reset invalid customer/supplier field
-        doc = frappe.get_doc("Item Price", d[0])
-        doc.save()
\ No newline at end of file
+    """
+    set proper customer and supplier details for item price
+    based on selling and buying values
+    """
+
+    # update for selling
+    frappe.db.sql(
+        """UPDATE `tabItem Price`
+        SET `reference` = `customer`, `supplier` = NULL
+        WHERE `selling` = 1
+        AND `buying` = 0
+        AND (`supplier` IS NOT NULL OR `supplier` = '')
+        AND `price_list` = `tabPrice List`.`name`
+        AND `tabPrice List`.`enabled` = 1""")
+
+    # update for buying
+    frappe.db.sql(
+        """UPDATE `tabItem Price``
+        SET `reference` = `supplier`, `customer` = NULL
+        WHERE `selling` = 0
+        AND `buying` = 1
+        AND (`supplier` IS NOT NULL OR `supplier` = '')
+        AND `price_list` = `tabPrice List`.`name`
+        AND `tabPrice List`.`enabled` = 1""")
```

Comment:
We should patch even if the price list is disable, so if the price is ever enabled the problem won't persists.

Reply:
No need. On saving thi item price document the fields will be corrected